doc/notebooks/jupyter_integration.ipynb

```diff
@@ -75,17 +75,35 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Let's display the DTM of our sample area to ensure all's working\n",
-    "\n",
+    "# Let's see what is in the example database\n",
+    "print(gs.read_command(\"g.list\", type=\"all\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "# Set computational region to the study area.\n",
-    "gs.parse_command(\"g.region\", raster=\"elevation\", flags=\"pg\")\n",
+    "gs.run_command(\"g.region\", raster=\"elevation\", flags=\"pg\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Demonstration of grassRenderer for non-interactive map display\n",
+    "spm_map = gj.GrassRenderer(height=600)\n",
+    "\n",
+    "spm_map.d_rast(raster=\"elevation\", color=\"grey\")\n",
```

Comment:
Would `spm_map.raster` be for new users more intuitive than `spm_map.d_rast` (or `spm_map.vector` instead of `spm_map.d_vect`)...?

Reply:
This is a great question. I think it would be easier for new users but it also goes against the GRASS convention and might be less intuitive for experienced GRASS users... I'd be curious to hear what others think. I can write it either way.

Reply:
I also think this is a great question. Even the name `GrassRenderer` is not necessary set. `spm_map` was little strange at first, but I think it is actually okay.

In a [GrassRenderer](https://github.com/ncsu-geoforall-lab/gis714-2021-tangible-landscape/blob/main/website/render_activities.py#L41) class I wrote for a different purpose, I had only a `run` method which just executed whatever command (like `run_command` does) since I wanted to pass any `d.*` commands which I was reading and [running programmatically](https://github.com/ncsu-geoforall-lab/gis714-2021-tangible-landscape/blob/main/website/render_activities.py#L233). I guess you could try to enforce that the command name must start with `d` so that it is not misused, but I'm not convinced it is needed.

`d_rast`, `d_vect` is a sure middle ground, but it requires either all modules to be mirrored or implementing the "resolve a missing attribute" magic as in PyGRASS module shortcuts to support all commands. It would be harder to implement what I did above using just this because you need to translate what you are reading to function calls, but it is doable in Python and perhaps a fringe use case in Jupyter.

`vector`, `raster` may work well as a hand-picked subset in addition to a more general interface or if you conceptualize the whole thing as a "display" (`display = GrassDisplay()`) where d.vect translates to `display.vector()`. Yet another approach is to have these as a common API with the folium-based renderer in parallel with the `d_vect` style.

Reply:
What about module shortcuts here?
https://grass.osgeo.org/grass79/manuals/libpython/pygrass.modules.html?highlight=shortcut#module-pygrass.modules.shortcuts

If there is value in making the API as much as possible compatibel with e.g. `folium` that should probably go into consideration for the non-interactive API as well?

I have to admitt that I am not familiar with folium. But for passing vector data to folium, that we discussed earlier, [viter](https://grass.osgeo.org/grass78/manuals/libpython/pygrass.vector.html?highlight=viter#pygrass.vector.VectorTopo.viter) from pygrass maybe an option too? There you could request only geometries within the map window of folium with the [find]() method.

Reply:
Yep, module shortcuts is what I meant by the "resolve a missing attribute" magic.

I suggest to:

1. Add `run(name, **kwargs)` now to:
    a. Support any command right away.
    b. Have a single place where the actual subprocess call is handled (i.e., use this method from the other methods).
2. Go with `d_rast` and `d_vect` or `raster` and `vector` (whatever seems better to you @chaedri) in this PR to have a nice interface.
3. Later, in another PR, adding more (all) `d_vect`-like methods later either explicitly defined, dynamically generated (`self.d_vect = some_locally_defined_function`), or through custom, simpler version of module shortcuts. I'm saying `d_vect`-like in a broad sense: they could be done like `display.vect` similarly to what I mentioned above. However, I think closer to module names, i.e., `d_vect` makes more sense here as it will clearly translate to all display module names.
4. Leave the compatibility with interactive display for later (again another PR) when the interactive display is somewhat finished. I think switching between displays just by changing the class name but not the rest of your code would be nice.